testing/test_terminal.py

```diff
@@ -2614,3 +2614,123 @@ def test_format_trimmed() -> None:
 
     assert _format_trimmed(" ({}) ", msg, len(msg) + 4) == " (unconditional skip) "
     assert _format_trimmed(" ({}) ", msg, len(msg) + 3) == " (unconditional ...) "
+
+
+def test_fine_grained_test_case_verbosity(pytester: Pytester):
+    p = pytester.makepyfile(_fine_grained_verbosity_file_contents())
+    pytester.makeini(
+        """
+        [pytest]
+        verbosity_test_cases = 2
+        """
+    )
+    result = pytester.runpytest(p)
+
+    result.stdout.fnmatch_lines(
+        [
+            f"{p.name}::test_ok PASSED                 [ 14%]",
+            f"{p.name}::test_words_fail FAILED         [ 28%]",
+            f"{p.name}::test_numbers_fail FAILED       [ 42%]",
+            f"{p.name}::test_long_text_fail FAILED     [ 57%]",
+            f"{p.name}::test_parametrize_fail[hello-1] FAILED [ 71%]",
+            f"{p.name}::test_parametrize_fail[world-987654321] FAILED [ 85%]",
+            f"{p.name}::test_sample_skip SKIPPED (some",
+            "long skip reason that will not fit on a single line with other content",
+            "that goes on and on and on and on and on)                                [100%]",
+        ],
+        consecutive=True,
+    )
+
+
+def test_fine_grained_test_case_verbosity_collect_only_negative_2(pytester: Pytester):
+    p = pytester.makepyfile(_fine_grained_verbosity_file_contents())
+    pytester.makeini(
+        """
+        [pytest]
+        verbosity_test_cases = -2
+        """
+    )
+    result = pytester.runpytest("--collect-only", p)
+
+    result.stdout.fnmatch_lines(
+        [
+            "collected 7 items",
+            "",
+            f"{p.name}: 7",
+        ],
+        consecutive=True,
+    )
+
+
+def test_fine_grained_test_case_verbosity_collect_only_positive_2(pytester: Pytester):
+    p = pytester.makepyfile(_fine_grained_verbosity_file_contents())
+    pytester.makeini(
+        """
+        [pytest]
+        verbosity_test_cases = 2
+        """
+    )
+    result = pytester.runpytest("--collect-only", p)
+
+    result.stdout.fnmatch_lines(
+        [
+            "collected 7 items",
+            "",
+            f"<Module {p.name}>",
+            "  <Function test_ok>",
+            "    some docstring",
+            "  <Function test_words_fail>",
+            "  <Function test_numbers_fail>",
+            "  <Function test_long_text_fail>",
+            "  <Function test_parametrize_fail[hello-1]>",
+            "  <Function test_parametrize_fail[world-987654321]>",
+            "  <Function test_sample_skip>",
+        ],
+        consecutive=True,
+    )
+
+
+def _fine_grained_verbosity_file_contents() -> str:
+    long_text = "Lorem ipsum dolor sit amet " * 10
+    return f"""
```

Comment:
Not sure we need this complex tests, just a simple parametrized test would allow us to test the purpose of the flag just as well, right?

```python
@pytest.mark.parametrize("i", range(7))
def test_ok(i): pass
```

The rationale is that we are not testing the details of the output, only that it is more or less verbose, so the names and the length of the tests docstrings do not plan a role here, and they are already being tested elsewhere.